docs/compose/Modal/README.md

```diff
@@ -0,0 +1,104 @@
+
+## DefaultModal
+
+| Day | Night | Right to Left |
+| --- | --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-action.png" alt="DefaultModal" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-action_dm.png" alt="DefaultModal - dark mode" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-action_rtl.png" alt="DefaultModal - rtl mode" width="375" /> |
+
+## ModalWithoutAction
+
+| Day | Night | Right to Left |
+| --- | --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-no-action.png" alt="ModalWithoutAction" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-no-action_dm.png" alt="ModalWithoutAction - dark mode" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-title-no-action_rtl.png" alt="ModalWithoutAction - rtl mode" width="375" /> |
+
+## ModalWithoutActionAndTitle
+
+| Day | Night | Right to Left |
+| --- | --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-no-title-no-action.png" alt="ModalWithoutActionAndTitle" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-no-title-no-action_dm.png" alt="ModalWithoutActionAndTitle - dark mode" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-no-title-no-action_rtl.png" alt="ModalWithoutActionAndTitle - rtl mode" width="375" /> |
+
+## ModalWithBackIcon
+
+| Day | Night | Right to Left |
+| --- | --- | --- |
+| <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-back-icon.png" alt="ModalWithBackIcon" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-back-icon_dm.png" alt="ModalWithBackIcon - dark mode" width="375" /> | <img src="https://raw.githubusercontent.com/Skyscanner/backpack-android/main/docs/compose/Modal/screenshots/modal-back-icon_rtl.png" alt="ModalWithBackIcon - rtl mode" width="375" /> |
+
+## Installation
+
+Backpack Compose is available through [Maven Central](https://search.maven.org/artifact/net.skyscanner.backpack/backpack-compose). Check the main [Readme](https://github.com/skyscanner/backpack-android#installation) for a complete installation guide.
+
+## Usage
+
+Example of a Modal with title, text action and text content:
```

Comment:
i think rather than having all three versions it should be enough just having one that sets all of them. however i would also include the state property with the remember function + the `openModal` flag (check how the readme for the bottom sheet is done for inspiration).
ideally we also call the hide function in the nav icon action to showcase that + resetting the `openModal` flag in the `onDismiss` callback. we might also want to explain why to use one or the other (ie the state will run the animation, the `onDismiss` will get called once the animation is completed and the modal is actually dimissed